File: minhas-financas-back-end/MinhasFinancas.Domain/Entities/PlanoEstrategicoFinanceiro.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MinhasFinancas.Domain.Entities
{
    public class PlanoEstrategicoFinanceiro
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid PlanoRaizId { get; set; }

        [ForeignKey(nameof(UsuarioId))]
        public string UsuarioId { get; set; } = string.Empty;

        [MaxLength(200)]
        public string Nome { get; set; } = string.Empty;

        [MaxLength(4000)]
        public string? Descricao { get; set; }

        [MaxLength(4000)]
        public string? Observacao { get; set; }

        public int NumeroVersao { get; set; } = 1;

        public DateTime DataInicioVigencia { get; set; }

        public DateTime? DataFimVigencia { get; set; }

        public DateTime DataCadastro { get; set; } = DateTime.UtcNow;

        public DateTime DataAtualizacao { get; set; } = DateTime.UtcNow;

        public bool Ativo { get; set; } = true;

        public Usuario? Usuario { get; set; }

        public List<ObjetivoPlanoEstrategico> Objetivos { get; set; } = [];
    }
}

File: minhas-financas-front-end/src/types/plano-estrategico-financeiro.ts
export enum PrioridadeObjetivoPlanoEstrategico {
  Baixa = 0,
  Media = 1,
  Alta = 2,
  Critica = 3,
}

export enum StatusObjetivoPlanoEstrategico {
  Planejado = 0,
  EmAndamento = 1,
  Concluido = 2,
  Cancelado = 3,
}

export interface ObjetivoPlanoEstrategicoPayload {
  id?: string | null;
  titulo: string;
  descricao?: string | null;
  prioridade: PrioridadeObjetivoPlanoEstrategico;
  status: StatusObjetivoPlanoEstrategico;
  ordem: number;
  dataAlvo?: string | null;
  valorAlvo?: number | null;
  valorAtual?: number | null;
  observacao?: string | null;
}

export interface SalvarPlanoEstrategicoFinanceiroPayload {
  nome: string;
  descricao?: string | null;
  observacao?: string | null;
  dataInicioVigencia?: string | null;
  objetivos: ObjetivoPlanoEstrategicoPayload[];
}

export interface ObjetivoPlanoEstrategico {
  id?: string | null;
  titulo: string;
  descricao?: string | null;
  prioridade: PrioridadeObjetivoPlanoEstrategico;
  status: StatusObjetivoPlanoEstrategico;
  ordem: number;
  dataAlvo?: string | null;
  valorAlvo?: number | null;
  valorAtual?: number | null;
  observacao?: string | null;
}

export interface PlanoEstrategicoFinanceiroResumo {
  id: string;
  planoRaizId: string;
  nome: string;
  descricao?: string | null;
  numeroVersao: number;
  dataInicioVigencia: string;
  dataFimVigencia?: string | null;
  dataCadastro: string;
  dataAtualizacao: string;
  ativo: boolean;
  quantidadeObjetivos: number;
}

export interface PlanoEstrategicoFinanceiroDetalhe {
  id: string;
  planoRaizId: string;
  nome: string;
  descricao?: string | null;
  observacao?: string | null;
  numeroVersao: number;
  dataInicioVigencia: string;
  dataFimVigencia?: string | null;
  dataCadastro: string;
  dataAtualizacao: string;
  ativo: boolean;
  objetivos: ObjetivoPlanoEstrategico[];
}
